### PR TITLE
vv issue #440

### PR DESCRIPTION
**Pull request by vince-vic**
## Description
[CM-1217](https://adazzle.atlassian.net/browse/CM-1217?filter=-1) Editors should close when clicked outside of them.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/adazzle/react-data-grid/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)
Currently the editor will only close when you click on the cell outside of it. The expected behaviour is for it to close when clicked anywhere outside of it.

**What is the new behavior?**
Editors close when clicked outside of them.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by richardware** [comment]
Hi, @vince-vic. Can you please complete the template above so that we have a better idea what you're working on?

**Comment by vince-vic** [comment]
@malonecj please review.

**Comment by vince-vic** [comment]
@malonecj please review.

**Comment by malonecj** [comment]
Great work :+1:

File: src/addons/editors/EditorContainer.js [hunk 1]
```diff
@@ -259,10 +273,15 @@ const EditorContainer = React.createClass({
   },
 
   render(): ?ReactElement {
+    let divStyle = {
```
Comment (malonecj) [inline src/addons/editors/EditorContainer.js]: any reason why we are changing the styles here like this?

File: src/addons/editors/EditorContainer.js [hunk 2]
```diff
@@ -259,10 +273,15 @@ const EditorContainer = React.createClass({
...
+      display: 'inline-block',
+      width: '200px',
+      height: '200px'
+    };
     return (
-        <div className={this.getContainerClass()} onKeyDown={this.onKeyDown} commit={this.commit}>
-        {this.createEditor()}
-        {this.renderStatusIcon()}
+        <div style={divStyle} className={this.getContainerClass()} onBlur={this.handleBlur} onKeyDown={this.onKeyDown} commit={this.commit}>
```
Comment (bird512) [inline src/addons/editors/EditorContainer.js]: I have a editor renderer with date picker. As the data picker popup is a single div and when try to open the popup, the onBlur will be called and this.commit will prevent the popup. Can we make this onBlur configable?

**Pull request merged**